core/model/modx/processors/system/phpthumb.class.php

```
@@ -55,6 +55,13 @@ public function process() {
         $src = $this->source->prepareSrcForThumb($src);
         if (empty($src)) return '';
 
+        if (strtolower(pathinfo($src, PATHINFO_EXTENSION)) === 'svg') {
+            /* Skip thumbnail generation for svg and output the file directly */
+            header('Content-Type: image/svg+xml');
+            echo @file_get_contents($src);
```

Why did it become necessary to silence errors? Can these errors occur?

---

> An E_WARNING level error is generated if filename cannot be found, length is less than zero, or if seeking to the specified offset in the stream fails.

I imagine this can happen when using an URL instead of a file from the media browser, or when the local file get deleted. Usually I silence these warnings because the result needs to be validated anyway.

In this case the suppression can be removed because warnings are already handled by the manager.